cmd/notary/keys.go

```diff
@@ -258,24 +267,43 @@ func (k *keyCommander) keysExportRoot(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	cs := cryptoservice.NewCryptoService("", ks...)
+
+	// Search for this key in all of our keystores, determine whether this key has a GUN
+	keyGun := ""
+	keyRole := ""
+	for _, store := range ks {
+		for keypath, role := range store.ListKeys() {
+			if filepath.Base(keypath) == keyID {
+				keyRole = role
+				if role != data.CanonicalRootRole {
+					dirPath := filepath.Dir(keypath)
```

Comment:
I would remove the indentation by changing this to `if role == data.CanonicalRootRole`